catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/RefreshRegistryEntries.java

```diff
@@ -68,10 +77,31 @@
 
     private boolean enableDelete = true;
 
+    private boolean initialized = false;
+
+    private ScheduledExecutorService executor;
+
+    private int taskWaitTimeSeconds = 30;
+
+    private int refreshInterval = 30;
+
+    private Future scheduledTask;
+
     public RefreshRegistryEntries() {
 
     }
 
+    public void init() {
+        scheduledTask = executor.scheduleWithFixedDelay(() -> {
+            try {
+                refreshRegistryEntries();
+            } catch (FederationAdminException e) {
+                LOGGER.debug("Problem refreshing registry entries.", e);
+            }
+        }, 10, refreshInterval, TimeUnit.SECONDS);
```

Comment:
Nothing. According to the javadocs once the task has finished (however long it takes) it will then wait the refreshInterval amount of time before running the task again.